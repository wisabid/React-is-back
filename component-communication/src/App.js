import React, { Component } from 'react';
import logo from './logo.svg';
import './App.css';
import Parent from './components/parent';

class App extends Component {
  constructor() {
    super();
    console.log('Constructor...')
    this.state = {
      title : "Propagate Me Alfie!"
    }
  }
  componentWillMount() {
    console.log('componentWillMount...')
  }

  componentDidMount() {
    console.log('component Did Mount...')
  }

  componentWillReceiveProps() {
    console.log('component Will receive props...')
  }

  shouldComponentUpdate(nextState, nextProps) {
    console.log('component Will receive props...')
    return true
  }

  componentWillUpdate() {
    
    console.log('component Will Update...YOU SHOULD NEVER SET STATE')
    // this.setState({
    //   title : 'tht'
    // })
  }

  componentDidUpdate() {
    console.log('component Did Update...YOU SHOULD NEVER SET STATE')
  }

  changemyname = (newtitle) => {
    this.setState({
      title : newtitle
    })
  }

  dothis = (ths) => {
    this.setState({
      title : ths
    })
  }


  dothat = (tht) => {
    this.setState({
      title : tht
    })
  }
  render() {
    console.log('RENDERRRRRRRRRRRR')
    return (
      <div className="App">
        <header className="App-header">
          <img src={logo} className="App-logo" alt="logo" />
          <Parent dothis={this.dothis.bind(this, 'DO THIS MANN')} dothat={this.dothat.bind(this, 'PLEASE DO THAT BOY')} changeme={this.changemyname.bind(this, 'Alfie! you Are Awesome!')}>{this.state.title}</Parent>
        </header>
      </div>
    );
  }
}

export default App;
