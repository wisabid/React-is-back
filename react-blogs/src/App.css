.App {
  text-align: center;
}

.App .linkedin{
  color:#0073b1;
}

.App .blackey{
  color:#000;
}

.App-logo {
  animation: App-logo-spin infinite 20s linear;
  height: 40vmin;
}

.App-header {
  background-color: #282c34;
  min-height: 100vh;
  display: flex;
  flex-direction: column;
  align-items: center;
  justify-content: center;
  font-size: calc(10px + 2vmin);
  color: white;
}

.linkedin .App-link {
  color: #61dafb;
}

.blackey .App-link {
  color: #ccc;
}

@keyframes App-logo-spin {
  from {
    transform: rotate(0deg);
  }
  to {
    transform: rotate(360deg);
  }
}
ul {
  list-style: none;
}
/* Component Level CSS Styles */
.float-right {
  float:right;
}
.linkedin .tiny-font {
  color: #000;
}
.blackey .tiny-font {
  color: blue;
}
.tiny-font {
  font-size: 11px;
  cursor: pointer;
  font-weight: bold;
}
.linkedin .blog-item {
  background:#0073b1;
  color:#fff;
}
.blackey .blog-item {
  background:#ddd;
  color:#000;
}
.blog-item {
  /* background : #bbdefb; */
 
  padding : 10px;
  margin: 10px;
  border-radius: 5px;
  text-align: left;
  font-family: Verdana, Geneva, Tahoma, sans-serif;
  
  box-shadow: 0 2px 4px 0 rgba(0,0,0,0.16),0 2px 10px 0 rgba(0,0,0,0.12);
  animation: 0.3s  ease-in 1 render-animation;
}
.blog-item.edit-item {
  opacity: 0;
}
.blog-item.edit-item.active {
  opacity: 1;
  transition: opacity 4s ease-in;
}
.blog-item .icon-actions {
  float: right;  
}
.blog-item input, .blog-item textarea, input[type="text"] {
  border: none;
  border-radius: 2px;
  padding: 10px;
  margin: 10px 0;
  width: 98%;
}

.blog-item h3, .blog-item p
{
word-break: break-word;
}
.highight {
  border : 5px solid #ddd
}
.action-new {
  cursor: pointer;
  color: #000;
}
header {
  margin:0 10px;
}
#search-bar {
  border : 1px solid #0073b1;
  border-radius:10px;
}
.blog-item h3 {
  margin: 3px;
}
.blog-item p {
  font-size:14px;
  line-height: 25px;
  font-weight: normal;
  font-style: italic;
}
.error-container, .message-container {
  color: orange;
  font-family: 13px;
  font-style: italic;
  text-align: center

}

@keyframes render-animation {
  from {
    opacity: 0;
  }
  to {
    opacity: 1;
  }
}

.suggestions ul {
  margin:0;
  padding:0;
}
.suggestions ul li {
  margin: 10px 0;
  background: #fff;
  border-radius: 5px;
  color: #0073b1;
  padding: 10px;
}
/* .blog-item .no-ptr.tiny-font {
  cursor: none;
} */

/* .example-enter {
  opacity: 0.01;
}

.example-enter.example-enter-active {
  opacity: 1;
  transition: opacity 500ms ease-in;
}

.example-leave {
  opacity: 1;
}

.example-leave.example-leave-active {
  opacity: 0.01;
  transition: opacity 300ms ease-in;
} */

/* Component Level CSS Styles end here */
