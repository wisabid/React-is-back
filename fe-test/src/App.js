import React, { Component } from 'react';
import logo from './logo.svg';
import './App.css';
import Blogs from './components/Blogs';

class App extends Component {
  

  render() {
    return (
      <div className="App">
        <Blogs />
      </div>
    );
  }
}

export default App;
