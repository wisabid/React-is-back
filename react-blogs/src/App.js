import React, { Component } from 'react';
import logo from './logo.svg';
import './App.css';
import Blogs from './components/Blogs';
import ThemeContext from './context/theme-context';

class App extends Component {
  state = {
    theme : 'linkedin'
  }

  changeTheme = (theme) => {
    console.log('Changing Theme...');
    const themes = ['blackey', 'linkedin'];    
    const newtheme = (this.state.theme === 'linkedin')?'blackey':'linkedin';
    this.setState({
      theme : newtheme
    })
  }

  render() {
    return (
      <ThemeContext.Provider value={{
        theme : this.state.theme, 
        changeTheme: this.changeTheme
      }}>
        <div className="App">
          <Blogs />
        </div>
      </ThemeContext.Provider>
    );
  }
}

export default App;
