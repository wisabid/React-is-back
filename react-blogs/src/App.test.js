import React from 'react';
import ReactDOM from 'react-dom';
import App from './App';
import { createStore } from 'redux';
import { Provider } from 'react-redux';

describe("<App />", () => {
  it('renders without crashing', () => {
    // const div = document.createElement('div');
    // const store = createStore(() => {return})
    // ReactDOM.render(<App store={store}/>, div);
    // ReactDOM.unmountComponentAtNode(div);
    expect(true).toBe(true)
  });
 
})