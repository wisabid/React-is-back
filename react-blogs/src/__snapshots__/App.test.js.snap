// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Blog /> renders Blog component without crashing 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <UNDEFINED
    addblog={true}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "children": Array [
        null,
        <div
          className="blog-item"
        >
          <div
            className="float-right tiny-font"
          >
            <span
              id="cancelBtn"
            >
              Cancel
            </span>
             | 
            <span
              id="submitbtn"
            >
              Submit
            </span>
          </div>
          <input
            maxLength="200"
            name="blogheader"
            onKeyUp={[Function]}
            placeholder="Header goes here"
            size="104"
            type="text"
          />
          <br />
          <textarea
            cols="95"
            name="blogbody"
            placeholder="Body goes here"
            rows="10"
          />
        </div>,
      ],
    },
    "ref": null,
    "rendered": Array [
      null,
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <div
              className="float-right tiny-font"
            >
              <span
                id="cancelBtn"
              >
                Cancel
              </span>
               | 
              <span
                id="submitbtn"
              >
                Submit
              </span>
            </div>,
            <input
              maxLength="200"
              name="blogheader"
              onKeyUp={[Function]}
              placeholder="Header goes here"
              size="104"
              type="text"
            />,
            <br />,
            <textarea
              cols="95"
              name="blogbody"
              placeholder="Body goes here"
              rows="10"
            />,
          ],
          "className": "blog-item",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <span
                  id="cancelBtn"
                >
                  Cancel
                </span>,
                " | ",
                <span
                  id="submitbtn"
                >
                  Submit
                </span>,
              ],
              "className": "float-right tiny-font",
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "Cancel",
                  "id": "cancelBtn",
                  "onClick": undefined,
                },
                "ref": null,
                "rendered": "Cancel",
                "type": "span",
              },
              " | ",
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "Submit",
                  "id": "submitbtn",
                  "onClick": undefined,
                },
                "ref": null,
                "rendered": "Submit",
                "type": "span",
              },
            ],
            "type": "div",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "maxLength": "200",
              "name": "blogheader",
              "onChange": undefined,
              "onKeyUp": [Function],
              "placeholder": "Header goes here",
              "size": "104",
              "type": "text",
            },
            "ref": [Function],
            "rendered": null,
            "type": "input",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {},
            "ref": null,
            "rendered": null,
            "type": "br",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "cols": "95",
              "name": "blogbody",
              "onChange": undefined,
              "placeholder": "Body goes here",
              "rows": "10",
            },
            "ref": [Function],
            "rendered": null,
            "type": "textarea",
          },
        ],
        "type": "div",
      },
    ],
    "type": Symbol(react.fragment),
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "children": Array [
          null,
          <div
            className="blog-item"
          >
            <div
              className="float-right tiny-font"
            >
              <span
                id="cancelBtn"
              >
                Cancel
              </span>
               | 
              <span
                id="submitbtn"
              >
                Submit
              </span>
            </div>
            <input
              maxLength="200"
              name="blogheader"
              onKeyUp={[Function]}
              placeholder="Header goes here"
              size="104"
              type="text"
            />
            <br />
            <textarea
              cols="95"
              name="blogbody"
              placeholder="Body goes here"
              rows="10"
            />
          </div>,
        ],
      },
      "ref": null,
      "rendered": Array [
        null,
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <div
                className="float-right tiny-font"
              >
                <span
                  id="cancelBtn"
                >
                  Cancel
                </span>
                 | 
                <span
                  id="submitbtn"
                >
                  Submit
                </span>
              </div>,
              <input
                maxLength="200"
                name="blogheader"
                onKeyUp={[Function]}
                placeholder="Header goes here"
                size="104"
                type="text"
              />,
              <br />,
              <textarea
                cols="95"
                name="blogbody"
                placeholder="Body goes here"
                rows="10"
              />,
            ],
            "className": "blog-item",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  <span
                    id="cancelBtn"
                  >
                    Cancel
                  </span>,
                  " | ",
                  <span
                    id="submitbtn"
                  >
                    Submit
                  </span>,
                ],
                "className": "float-right tiny-font",
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "Cancel",
                    "id": "cancelBtn",
                    "onClick": undefined,
                  },
                  "ref": null,
                  "rendered": "Cancel",
                  "type": "span",
                },
                " | ",
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "Submit",
                    "id": "submitbtn",
                    "onClick": undefined,
                  },
                  "ref": null,
                  "rendered": "Submit",
                  "type": "span",
                },
              ],
              "type": "div",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "maxLength": "200",
                "name": "blogheader",
                "onChange": undefined,
                "onKeyUp": [Function],
                "placeholder": "Header goes here",
                "size": "104",
                "type": "text",
              },
              "ref": [Function],
              "rendered": null,
              "type": "input",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {},
              "ref": null,
              "rendered": null,
              "type": "br",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "cols": "95",
                "name": "blogbody",
                "onChange": undefined,
                "placeholder": "Body goes here",
                "rows": "10",
              },
              "ref": [Function],
              "rendered": null,
              "type": "textarea",
            },
          ],
          "type": "div",
        },
      ],
      "type": Symbol(react.fragment),
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;

exports[`<Blogs /> <Blogs /> should render correctly 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Blogs />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "children": Array [
        <header>
          <h2>
            B L O G S (
            <span
              className="action-new"
              onClick={[Function]}
              title="Add New"
            >
              +
            </span>
            )
          </h2>
          <span>
            <input
              id="search-bar"
              onChange={[Function]}
              placeholder="Search text"
              type="text"
              value=""
            />
          </span>
        </header>,
        <div
          className="blog-item message-container"
          style={
            Object {
              "color": "lightgreen",
            }
          }
        >
          No Search Results to display !
        </div>,
        <UNDEFINED
          fallback={
            <h1>
              Loading...
            </h1>
          }
        />,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <h2>
              B L O G S (
              <span
                className="action-new"
                onClick={[Function]}
                title="Add New"
              >
                +
              </span>
              )
            </h2>,
            <span>
              <input
                id="search-bar"
                onChange={[Function]}
                placeholder="Search text"
                type="text"
                value=""
              />
            </span>,
          ],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                "B L O G S (",
                <span
                  className="action-new"
                  onClick={[Function]}
                  title="Add New"
                >
                  +
                </span>,
                ")",
              ],
            },
            "ref": null,
            "rendered": Array [
              "B L O G S (",
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "+",
                  "className": "action-new",
                  "onClick": [Function],
                  "title": "Add New",
                },
                "ref": null,
                "rendered": "+",
                "type": "span",
              },
              ")",
            ],
            "type": "h2",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": <input
                id="search-bar"
                onChange={[Function]}
                placeholder="Search text"
                type="text"
                value=""
              />,
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "id": "search-bar",
                "onChange": [Function],
                "placeholder": "Search text",
                "type": "text",
                "value": "",
              },
              "ref": null,
              "rendered": null,
              "type": "input",
            },
            "type": "span",
          },
        ],
        "type": "header",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "No Search Results to display !",
          "className": "blog-item message-container",
          "style": Object {
            "color": "lightgreen",
          },
        },
        "ref": null,
        "rendered": "No Search Results to display !",
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "children": Array [],
          "fallback": <h1>
            Loading...
          </h1>,
        },
        "ref": null,
        "rendered": Array [],
        "type": Symbol(react.suspense),
      },
    ],
    "type": Symbol(react.fragment),
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "children": Array [
          <header>
            <h2>
              B L O G S (
              <span
                className="action-new"
                onClick={[Function]}
                title="Add New"
              >
                +
              </span>
              )
            </h2>
            <span>
              <input
                id="search-bar"
                onChange={[Function]}
                placeholder="Search text"
                type="text"
                value=""
              />
            </span>
          </header>,
          <div
            className="blog-item message-container"
            style={
              Object {
                "color": "lightgreen",
              }
            }
          >
            No Search Results to display !
          </div>,
          <UNDEFINED
            fallback={
              <h1>
                Loading...
              </h1>
            }
          />,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <h2>
                B L O G S (
                <span
                  className="action-new"
                  onClick={[Function]}
                  title="Add New"
                >
                  +
                </span>
                )
              </h2>,
              <span>
                <input
                  id="search-bar"
                  onChange={[Function]}
                  placeholder="Search text"
                  type="text"
                  value=""
                />
              </span>,
            ],
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  "B L O G S (",
                  <span
                    className="action-new"
                    onClick={[Function]}
                    title="Add New"
                  >
                    +
                  </span>,
                  ")",
                ],
              },
              "ref": null,
              "rendered": Array [
                "B L O G S (",
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "+",
                    "className": "action-new",
                    "onClick": [Function],
                    "title": "Add New",
                  },
                  "ref": null,
                  "rendered": "+",
                  "type": "span",
                },
                ")",
              ],
              "type": "h2",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": <input
                  id="search-bar"
                  onChange={[Function]}
                  placeholder="Search text"
                  type="text"
                  value=""
                />,
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "id": "search-bar",
                  "onChange": [Function],
                  "placeholder": "Search text",
                  "type": "text",
                  "value": "",
                },
                "ref": null,
                "rendered": null,
                "type": "input",
              },
              "type": "span",
            },
          ],
          "type": "header",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "No Search Results to display !",
            "className": "blog-item message-container",
            "style": Object {
              "color": "lightgreen",
            },
          },
          "ref": null,
          "rendered": "No Search Results to display !",
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "children": Array [],
            "fallback": <h1>
              Loading...
            </h1>,
          },
          "ref": null,
          "rendered": Array [],
          "type": Symbol(react.suspense),
        },
      ],
      "type": Symbol(react.fragment),
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
  Symbol(enzyme.__childContext__): null,
}
`;
